### PR TITLE
refactor: remove `tenv` from `Resolver`

### PR DESCRIPTION
**Pull request by mlutze**
An experiment. The `tenv` appears useless, so we will see what happens when we take it away.

### CODE REVIEW

**Comment by magnus-madsen** [comment]
But That's ...

**Comment by magnus-madsen** [comment]
image

**Comment by magnus-madsen** [comment]
Before I press the button, is there any intuition?

File: main/src/ca/uwaterloo/flix/language/phase/Resolver.scala [hunk 1]
```diff
@@ -558,15 +551,13 @@ object Resolver {
...
-      def visitExp(e0: NamedAst.Expression, tenv0: Map[Symbol.VarSym, Type], region: Option[Symbol.VarSym]): Validation[ResolvedAst.Expression, ResolutionError] = e0 match {
+      def visitExp(e0: NamedAst.Expression, region: Option[Symbol.VarSym]): Validation[ResolvedAst.Expression, ResolutionError] = e0 match {
 
         case NamedAst.Expression.Wild(loc) =>
           ResolvedAst.Expression.Wild(loc).toSuccess
 
-        case NamedAst.Expression.Var(sym, loc) => tenv0.get(sym) match {
-          case None => ResolvedAst.Expression.Var(sym, sym.tvar, loc).toSuccess
-          case Some(tpe) => ResolvedAst.Expression.Var(sym, tpe, loc).toSuccess
-        }
```
Comment (mlutze) [inline main/src/ca/uwaterloo/flix/language/phase/Resolver.scala:569]: This is the one place it is used.

**Comment by mlutze** [comment]
The intuition is that the tenv is redundant. We use it in one place to lookup the tvar of a VarSym. But the VarSym already contains its own tvar, so keeping a separate map is useless.

File: main/src/ca/uwaterloo/flix/language/phase/Resolver.scala (hunk 1 above)
Comment (magnus-madsen) [reply main/src/ca/uwaterloo/flix/language/phase/Resolver.scala:569]: Roger.

**Pull request merged**